sklearn/metrics/pairwise.py

```diff
@@ -654,19 +660,44 @@ def pairwise_distances_argmin_min(
     """
     X, Y = check_pairwise_arrays(X, Y)
 
-    if metric_kwargs is None:
-        metric_kwargs = {}
-
     if axis == 0:
         X, Y = Y, X
 
-    indices, values = zip(
-        *pairwise_distances_chunked(
-            X, Y, reduce_func=_argmin_min_reduce, metric=metric, **metric_kwargs
+    if metric_kwargs is None:
+        metric_kwargs = {}
+
+    if PairwiseDistancesArgKmin.is_usable_for(X, Y, metric):
+        # This is an adaptation for one "sqeuclidean" specification.
```

Comment:
What do you mean by "adaption"? I don't get what you'd like to say.

Reply:
This is an adaptor. But I guess this comment can be removed.